### DISCUSSION TITLE
How to filter out pattern not found errors?

### DISCUSSION THREAD

**Discussion by prescientmoon** [folke/noice.nvim#204] [Q&A]
I have tried adding "c" to shortmess but that doesn't do anything...

Additional question: why do I sometimes see "/whatever" as a message when searching that? Is there any way to remove that?

**Comment by folke** [comment]
Your answer is likely in https://github.com/folke/noice.nvim/wiki/A-Guide-to-Messages

**Reply by prescientmoon** [reply]
@folke Yep, that's where I got the "adding c to shortmess" idea from, but it doesn't seem to work...

**Reply by folke** [reply]
I think you need the `s` option

**Reply by folke** [reply]
The `c` flag is for insert completion, but you need the flag for search.

**Reply by prescientmoon** [reply]
I have also added "s" and getting the same result :thinking:

**Reply by folke** [reply]
Then you must not be setting it right. Noice has no control over those messages. That's all Neovim. I have added `s` and others and it works as intended for me.

**Comment by dimfeld** [comment]
I just starting using noice and encountered this same problem. Adding `s` to `shortmess` didn't seem to work for me either, for the specific "Pattern not found" error message. (c.f. https://github.com/neovim/neovim/issues/24059, there may be an interaction with `cmdheight=0` here)

The following `routes` config can hide it though.

Note that the second condition here will hide any message starting with "/", not just the message echoed when searching for a nonexistent string, but I couldn't find something more specific to filter on.

```lua
      routes = {
        {
          opts = { skip = true },
          filter = {
            any = {
              {
                error = true,
                find = "E486: Pattern not found",
              },
              {
                event = "msg_show",
                cond = function(msg)
                  return msg:content():sub(1, 1) == "/"
                end,
              },
            },
          },
        },
      },
```
